exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -329,6 +330,13 @@ func (prwe *prwExporter) execute(ctx context.Context, writeReq *prompb.WriteRequ
 		if resp.StatusCode >= 500 && resp.StatusCode < 600 {
 			return rerr
 		}
+
+		// 429 errors are recoverable and the exporter should retry if RetryOnHTTP429 enabled
+		// Reference: https://github.com/prometheus/prometheus/pull/12677
+		if prwe.exporterSettings.RetryOnHTTP429 && resp.StatusCode == 429 {
+			return rerr
```

Comment:
I think this needs to handle the `Retry-After` header so that we can respect the server telling us when it wants the next retry to happen.

Reply:
We discussed this on one of the otel prometheus WG, and wanted to start with the simpler implementation of just retrying with the defaults, and later improving by adding support for that, just to get something out the door people can try, is that approach okay with you?

Reply:
As discussed in the WG today we're agreed to move forward without supporting `Retry-After` while this feature is behind an alpha-level feature gate.